stmhal/mphalport.h

```diff
@@ -6,10 +6,16 @@
 // go in some MCU-specific header, but for now it lives here.
 #if defined(MCU_SERIES_F4)
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1fff7a10)
+#define MP_HAL_CLEANINVALIDATECACHE(addr, size)
+#define MP_HAL_CLEANCACHE(addr, size)
 #elif defined(MCU_SERIES_F7)
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1ff0f420)
+#define MP_HAL_CLEANINVALIDATECACHE(addr, size) (SCB_CleanInvalidateDCache_by_Addr((uint32_t*)(((uint32_t)addr ) & ~(0x1F)), (size) + ((uint32_t)addr - (((uint32_t)addr ) & ~(0x1F)))))
```

Comment:
I'm assuming that size needs to be aligned as well, so this probably needs to be something like:
```
((addr + size + 0x1f) & ~0x1f) - (addr & ~0x1f)
```
(addr + size + 0x1f) & ~0x1f will calculate an aligned end-ptr and (addr & 0x1f) calculates an aligned start ptr.